### PR TITLE
Add frontier (EVM) integration.

### PR DESCRIPTION
**Pull request by jnaviask**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[submodule "vendor/frontier"]
+	path = vendor/frontier
+	url = https://github.com/PureStake/frontier.git
+	branch = joshy-parachains-v1
```
Comment (drewstone) [inline .gitmodules:4]: this branch name seems indicative of some features we may not want included, though might be for name's sake. Do we know when this ought to be merged and what its contents are?

File: node/runtime/src/lib.rs [hunk 2]
```diff
@@ -872,6 +882,177 @@ impl pallet_contracts::Config for Runtime {
...
+	gas_transaction_zero_data: 4,
+	gas_transaction_non_zero_data: 16,
+	sstore_gas_metering: true,
+	sstore_revert_under_stipend: true,
+	err_on_call_with_more_gas: false,
+	empty_considered_exists: false,
+	create_increase_nonce: true,
+	call_l64_after_gas: true,
+	stack_limit: 1024,
+	memory_limit: usize::max_value(),
```
Comment (drewstone) [inline node/runtime/src/lib.rs:923]: seems ominous, we should review these parameters as well and try to compare against ETH as well as other people's impl

File: node/runtime/src/lib.rs [hunk 3]
```diff
@@ -872,6 +882,177 @@ impl pallet_contracts::Config for Runtime {
...
+	use pallet_evm::Precompile;
+
+	// identity
+	let addr_ecrecover = H160::from_str("0000000000000000000000000000000000000001").expect("Invalid address at precompiles generation");
+	// ecrecover
+	let addr_sha256 = H160::from_str("0000000000000000000000000000000000000002").expect("Invalid address at precompiles generation");
+	// sha256
+	let addr_ripemd160 = H160::from_str("0000000000000000000000000000000000000003").expect("Invalid address at precompiles generation");
+	// ripemd160
+	let _addr_modexp = H160::from_str("0000000000000000000000000000000000000004").expect("Invalid address at precompiles generation");
```
Comment (drewstone) [inline node/runtime/src/lib.rs:987]: can we remove all these precompiles under ripemd160 for now. The PR never got merged into Substrate and don't exist yet.

File: node/runtime/Cargo.toml [hunk 4]
```diff
@@ -86,6 +87,10 @@ chainbridge = { path = "../../modules/chainbridge", default-features = false }
 orml-tokens = { git = "https://github.com/open-web3-stack/open-runtime-module-library", rev = "f69c13bf90cc007b553ce228108ad822c33f8459", default-features = false }
```
Comment (drewstone) [inline node/runtime/Cargo.toml:87]: might need a rebase or feel free to remove these orml libraries too if its not too much trouble.

**Review by drewstone — CHANGES_REQUESTED** [review]
Some small changes to make before closer inspection of parameters.

File: .gitmodules (hunk 1 above)
Comment (jnaviask) [reply .gitmodules:4]: It got merged earlier today so I can just switch back to default frontier.

File: node/runtime/src/lib.rs (hunk 2 above)
Comment (jnaviask) [reply node/runtime/src/lib.rs:923]: Yeah Acala has their own version -- we can use that as a reference. However keep in mind these params are me cloning the ISTANBUL_CONFIG from the source repo and only changing the contract creation limit.

File: node/runtime/Cargo.toml (hunk 4 above)
Comment (jnaviask) [reply node/runtime/Cargo.toml:87]: Why? They seem to build OK

Comment (drewstone) [reply node/runtime/Cargo.toml:87]: We aren't using them afaik? I removed any inclusion into the runtime for them in place of native assets.

File: .gitmodules (hunk 1 above)
Comment (jnaviask) [reply .gitmodules:4]: Also note that this doesn't mean anything -- what matters is the commit hash of the vendor subdirectory, which points to our fork of frontier.

File: node/runtime/src/lib.rs (hunk 2 above)
Comment (jnaviask) [reply node/runtime/src/lib.rs:923]: Here's their version: https://github.com/AcalaNetwork/Acala/blob/master/modules/evm/src/lib.rs#L61

Comment (jnaviask) [reply node/runtime/src/lib.rs:923]: Seems basically the same change I made + some gas return changes I'm not sure about.

**Pull request merged**